docs/design-documents/hal/0002-pinmap-extension.md

```diff
@@ -83,3 +83,5 @@ MBED_WEAK const PinList *pinmap_restricted_pins()
     return &pin_list;
 }
 ```
+
+In addition to `pinmap_restricted_pins()`, the `pinmap_restricted_pins_gpio()` is used to skip pins during GPIO testing. By default this function returns an empty list and any target can override it to provide a list of pins that should be skipped. For example, D14 and D15 pins present in the Arduino form factor of FRDM-K64F have fixed pull-up resistors. The `PullDown` `PinMode` is impossible to test. However, other peripherals like the I2C may be successfully tested on these pins.
```

Comment:
Its likely that more exclusion lists will be needed for various hardware quirks. What about something like `pinmap_pin_info(PinName pin, pin_info_t *info)` which returns a structure of info for the given pin? This could take the place of both `pinmap_restricted_pins()` and `pinmap_restricted_pins_gpio()` by having these as fields in the struct. As new types of hardware anomalies are found they could be added to the struct. The struct could look something like this (As an example):
```c
typedef struct {
    bool restricted;    // Pin should be avoided for all tests
    PullType pull;      // Pin has a hardware pullup or pulldown or is not pulled
    bool i2c_shared;    // One or more I2C devices may be connected to this pin
    bool spi_shared;    // One or more SPI devices may be connected to this pin
    bool needs_jumper;  // This pin is unconnected with default board jumper setting
} pin_info_t;
```

@maclobdell what are your thoughts on something like this?

Reply:
Sounds like a good idea. A weak `pinmap_pin_info()` would do the trick. This would require updating https://github.com/ARMmbed/fpga-ci-test-shield/blob/master/mbed-code/test_utils.h, @maciejbocianski what do you think?

Reply:
@c1728p9 what will be use case of i2c_shared?
Actually we have only problems with gpio/gpio_irq tests. Limitation of others (i2c/spi/qspi/...) are or should be covered in PinMap_XXX. I think that for now having `pinmap_restricted_pins_gpio` list with pulled pins is enough

Reply:
I agree with @maciejbocianski -- I don't see any need for more exclusion lists for now.